services/motion/builtin/move_request.go

```diff
@@ -140,6 +141,21 @@ func (mr *moveRequest) AnchorGeoPose() *spatialmath.GeoPose {
 // execute attempts to follow a given Plan starting from the index percribed by waypointIndex.
 // Note that waypointIndex is an atomic int that is incremented in this function after each waypoint has been successfully reached.
 func (mr *moveRequest) execute(ctx context.Context, plan motionplan.Plan) (state.ExecuteResponse, error) {
+	// Determine if we already are at the goal
+	// If our motion profile is position_only then, we only check against our current & desired position
+	// Conversely if our motion profile is anything else, then we also need to check again our
+	// current & desired orientation
+	if mr.motionProfile == motionplan.PositionOnlyMotionProfile {
```

Comment:
This is really the success criteria check. The `deviatedFromPlan` check at the end of this function should encapsulate this logic too and does not.  Rather than storing the motion profile in the move request we could store a success function similar to how we do `mr.position`